kotlinx-coroutines-core/native/src/CoroutineContext.kt

```diff
@@ -8,33 +8,29 @@ import kotlinx.coroutines.internal.*
 import kotlin.coroutines.*
 import kotlin.native.concurrent.*
 
-private fun takeEventLoop(): EventLoopImpl =
-    ThreadLocalEventLoop.currentOrNull() as? EventLoopImpl ?:
-        error("There is no event loop. Use runBlocking { ... } to start one.")
-
 internal actual object DefaultExecutor : CoroutineDispatcher(), Delay {
```

Comment:
I checked the uses of `DefaultExecutor`. It seems there are two places where it can be accessed in Native outside of `DefaultDelay`:
* `kotlinx-coroutines-core/common/src/EventLoop.common.kt`: `EventLoopImplBase.enqueue`.
* `kotlinx-coroutines-core/native/src/EventLoop.kt`: `reschedule`.

The only way these could be triggered, it seems, is if the event loop is closed, but new tasks arrive.

The only way for the event loop to be closed on Native is via a call to `decrementUseCount` if the event loop is shared, which it is when `runBlocking` is used.

So, it looks like it's possible to access this object even when `multithreadingSupported` is false.

Reply:
Fixed

Reply:
Sorry, I think I was right initially: `reschedule` is dangerous after all.